File: app/src/main/java/com/example/librarymanagement/adapter/FriendList.java
package com.example.librarymanagement.adapter;

import android.content.Context;
import android.view.*;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import com.example.librarymanagement.R;

import java.util.List;

public class FriendList extends RecyclerView.Adapter<FriendList.MyViewHolder>{
    private Context context;
    private List< String > list;
    private List<Integer> imagelist;
    private View inflater;
    //构造方法，传入数据
    public FriendList (Context context, List<String> list, List<Integer> imagelist){
        this.context = context;
        this.list = list;
        this.imagelist = imagelist;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        //创建ViewHolder，返回每一项的布局
        inflater = LayoutInflater.from(context).inflate(R.layout.friend_list, parent, false);
        MyViewHolder myViewHolder = new MyViewHolder(inflater);
        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        //将数据和控件绑定
        holder.textView.setText(list.get(position));
        holder.portrait.setImageResource(imagelist.get(position));
    }

    @Override
    public int getItemCount() {
        //返回Item总条数
        return list.size();
    }

    //内部类，绑定控件
    class MyViewHolder extends RecyclerView.ViewHolder{
        TextView textView;
        ImageView portrait;
        public MyViewHolder (View itemView) {
            super(itemView);
            textView = (TextView) itemView . findViewById (R.id.name);
            portrait = (ImageView) itemView . findViewById(R.id.portrait);
        }
    }
}

File: app/src/main/java/com/example/librarymanagement/adapter/FriendList.kt
package com.example.librarymanagement.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.librarymanagement.R

class FriendList : RecyclerView.Adapter<FriendList.MyHolder> {
    private var list: ArrayList<String>? = null
    private var imagelist: ArrayList<Int>? = null
    private var context: Context? = null
    private var itemClickListener: IKotlinItemClickListener? = null

    constructor (mContext: Context, list: ArrayList<String>?, imagelist: ArrayList<Int>?) {
        this.context = mContext
        this.list = list
        this.imagelist = imagelist
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        var view = LayoutInflater.from(context).inflate(R.layout.friend_list, parent, false)
        return MyHolder(view)
    }

    override fun getItemCount(): Int = list?.size!!

    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        holder?.text?.text = list!![position]
        holder?.image?.setImageResource(imagelist!![position])

        // 点击事件
        holder.itemView.setOnClickListener {
            itemClickListener!!.onItemClickListener(position)
        }

    }

    class MyHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {
        // !! 断言
        var text: TextView = itemView!!.findViewById(R.id.name)
        var image: ImageView = itemView!!.findViewById(R.id.portrait)
    }

    // 提供set方法
    fun setOnKotlinItemClickListener(itemClickListener: IKotlinItemClickListener) {
        this.itemClickListener = itemClickListener
    }

    //自定义接口
    interface IKotlinItemClickListener {
        fun onItemClickListener(position: Int)
    }

}
